test/NuGet.Core.FuncTests/NuGet.Protocol.FuncTest/AuthenticationHandlerTests.cs

```diff
@@ -170,6 +307,11 @@ public void HandleHttpRequest(IAsyncResult result)
                         else
                         {
                             context.Response.StatusCode = 200;
+
+                            var encodedCredentials = authorization.Substring("Basic ".Length).Trim();
+                            var credentials = Encoding.UTF8.GetString(Convert.FromBase64String(encodedCredentials));
+                            var split = credentials.Split(':');
+                            _capturedCredentials.Add(new NetworkCredential(userName: split[0], password: split[1]));
```

Comment:
Can the test use `Requests[i].Header["Authentication"]` instead or creating a second list?  This way tests that don't validate the credentials don't pay the cost of parsing the credential. Not that this minor perf cost in func tests really matters. I just feel it's weird that test infrastructure is spending effort on something that potentially only a subtest of tests using it will need.

If you think it's particularly useful to keep this captured credentials list, then I think that unauthenticated requests should insert `null`, so that `CapturedCredentials[i]` always corresponds to `Requests[i]`.

I'm thinking about it from an API design point of view, and what would make sense to other people who might want to write new tests and re-use the `CapturedCredentials` API.

Reply:
> Can the test use Requests[i].Header["Authentication"] instead or creating a second list?

Yup. I removed the CapturedCredentials collection from the API. I followed your suggestion use Requests[i].Header["Authentication"] to obtain the credentials from the Requests collection. Test passed locally, fixed in the latest commit.